src/box/alter.cc

```diff
@@ -2028,6 +2029,31 @@ filter_temporary_ddl_stmt(struct txn *txn, const struct space_def *def)
 	return 0;
 }
 
+/**
+ * The helper sets system trigger for a system space without surrogate space.
+ * If the space isn't system or has a surrogate space, it's no-op.
+ */
+static void
+on_replace_dd_space_set_system_trigger(struct space *space)
+{
+	if (space_id(space) == BOX_GC_CONSUMERS_ID) {
+		/*
+		 * Since the space can be re-created on upgrade, it's important
+		 * for each copy of space to have its own copy of trigger.
+		 * Otherwise two spaces may exist at the same time (e.g. applier
+		 * commits transactions asynchronously, so DDL dropping the
+		 * space won't be committed when the new copy is being created)
+		 * and dropping one of them (and the trigger as well) will lead
+		 * to crash.
+		 */
+		struct trigger *trigger =
+			(struct trigger *)xmalloc(sizeof(*trigger));
+		trigger_create(trigger, on_replace_dd_gc_consumers, NULL,
```

Comment:
I'd rather not try to incapsulate triggers into gc submodule and put them right into `alter`. Let's see, what @Gerold103 will say

Reply:
By the way, it was Vova idea to hide triggers in `gc.c`.

Reply:
There is no `on_replace` trigger anymore, so the comment is outdated.